### PR TITLE
feat: add new auth with magic.link

### PR DESCRIPTION
**Pull request by hugomrdias**

### CODE REVIEW

File: website/styles/global.css [hunk 1]
```diff
@@ -66,3 +66,22 @@
...
+  width: 100px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+.table-responsive {
+  display: block;
+  width: 100%;
+  overflow-x: auto;
+}
```
Comment (alanshaw) [inline website/styles/global.css:87]: These can almost all be tachyons classes, but can be fixed later.

File: website/components/navbar.js [hunk 2]
```diff
@@ -1,62 +1,58 @@
...
+  function logout() {
+    getMagic().user.logout()
+    setUser(undefined)
+    Router.push('/')
+  }
 
-export default function Navbar({ user, loginUrl, bgColor = 'nsorange' }) {
   return (
-    <nav className={`dn dn-m db-ns bg-${bgColor}`}>
+    <nav className={` bg-${bgColor}`}>
```
Comment (alanshaw) [inline website/components/navbar.js:16]: Those classes are there so the navbar disappears on mobile as it did not render well. We need a hamburger or something really. Could you open an issue to get it fixed?

**Review by alanshaw — CHANGES_REQUESTED** [review]
NICE! Some minor issues, but I think we can get this merged ASAP!

File: site/src/utils/auth.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+/**
+ *
+ * @param {FetchEvent} event
+ * @param {any} data
+ */
+export async function verify(event, data) {
+  const auth = event.request.headers.get('Authorization') || ''
+  const token = magic.utils.parseAuthorizationHeader(auth)
+
+  magic.token.validate(token)
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply site/src/utils/auth.js:58]: when we deploy if the user still has the old session valid, the frontend will ignore it and show the logged out UI so they always need to re-login back and they will get a fresh new token from magic

File: site/src/utils/auth.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+/**
+ *
+ * @param {*} data
+ * @param {import('@magic-sdk/admin').MagicUserMetadata} magicMetadata
+ * @returns {Promise<User>}
+ */
+async function parseGithub(data, magicMetadata) {
+  const sub = `github|${data.oauth.userHandle}`
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply site/src/utils/auth.js:80]: no, i will get the proper type in there

File: site/src/utils/jwt.js [hunk 5]
```diff
@@ -81,7 +81,8 @@ export async function verifyJWT(token, secret = auth0.salt, alg = 'HS256') {
   var tokenParts = token.split('.')
 
   if (tokenParts.length !== 3) {
-    throw new Error('token must have 3 parts')
+    return false
+    // throw new Error('token must have 3 parts')
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply site/src/utils/jwt.js:85]: if we throw here the `validate` method will be more complex

File: website/components/navbar.js (hunk 2 above)
Comment (hugomrdias) [reply website/components/navbar.js:16]: yes i will, i have tweak the navbar and its good enough for now in mobile

File: .github/workflows/client.yml [hunk 6]
```diff
@@ -67,34 +67,34 @@ jobs:
...
+  #     - name: Build client
+  #       run: yarn --cwd client build
+
+  #     - name: Publish client
+  #       uses: mikeal/merge-release@master
+  #       env:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  #         NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
+  #         SRC_PACKAGE_DIR: client
+  #         DEPLOY_DIR: client
```
Comment (alanshaw) [inline .github/workflows/client.yml:97]: Please create an issue to re-instate.

File: site/src/constants.js [hunk 7]
```diff
@@ -1,19 +1,15 @@
-// let AUTH0_DOMAIN, AUTH0_CLIENT_ID, AUTH0_CLIENT_SECRET, SALT, PINATA_JWT
+// let _MAGIC_SECRET_KEY, SALT, PINATA_JWT
```
Comment (alanshaw) [inline site/src/constants.js:1]:
```suggestion
// let MAGIC_SECRET_KEY, SALT, PINATA_JWT
```

File: client/src/lib.js [hunk 8]
```diff
@@ -44,7 +44,7 @@ class NFTStorage {
    *
    * @param {{token: string, endpoint?:URL}} options
    */
-  constructor({ token, endpoint = new URL('https://nft.storage') }) {
+  constructor({ token, endpoint = new URL('https://api.nft.storage') }) {
```
Comment (alanshaw) [inline client/src/lib.js:47]: ⚠️ We need to redirect the old URLs in cloudflare for old clients.

File: website/lib/api.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
+import { getMagic } from './magic'
+
+export const API = process.env.NEXT_PUBLIC_API
```
Comment (alanshaw) [inline website/lib/api.js:3]: Where is this value documented/set?

File: website/lib/magic.js [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+export function getMagic() {
+  if (typeof document === 'undefined') {
+    return undefined // static export
+  }
+
+  if (magic) {
+    return magic
+  }
+  magic = new Magic(process.env.NEXT_PUBLIC_MAGIC, {
```
Comment (alanshaw) [inline website/lib/magic.js:16]: Where is this value documented/set?

**Review by alanshaw — APPROVED** [review]
Just some final comments, nothing particularly blocking so I've approved, my main concern is `NEXT_PUBLIC_API` and `NEXT_PUBLIC_MAGIC` which are undocumented and I can't see where they're set.

**Comment by hugomrdias** [comment]
> Please create an issue to re-instate. 

https://github.com/ipfs-shipyard/nft.storage/issues/66

File: .github/workflows/client.yml (hunk 6 above)
Comment (hugomrdias) [reply .github/workflows/client.yml:97]: https://github.com/ipfs-shipyard/nft.storage/issues/66

File: client/src/lib.js (hunk 8 above)
Comment (hugomrdias) [reply client/src/lib.js:47]: already done for production

File: website/lib/api.js (hunk 9 above)
Comment (hugomrdias) [reply website/lib/api.js:3]: added docs in the readme

File: website/lib/magic.js (hunk 10 above)
Comment (hugomrdias) [reply website/lib/magic.js:16]: added docs in the readme

**Pull request merged**